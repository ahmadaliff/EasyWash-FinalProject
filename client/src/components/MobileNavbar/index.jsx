import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { useNavigate } from 'react-router-dom';

import {
  DryCleaningOutlined,
  FavoriteBorder,
  LayersClear,
  LibraryBooksOutlined,
  LocalLaundryServiceOutlined,
  MessageOutlined,
  PersonOutline,
  ShoppingCartOutlined,
} from '@mui/icons-material';
import { BottomNavigation, BottomNavigationAction } from '@mui/material';

import classes from '@components/MobileNavbar/style.module.scss';

const MobileNavbar = ({ login, user, intl: { formatMessage } }) => {
  const navigate = useNavigate();
  return (
    <div className={`${classes.MobileNavbar} ${!login && classes.hide}`} data-testid="navbar">
      <BottomNavigation showLabels>
        {user?.role === 'user' && (
          <BottomNavigationAction
            label="Favorit"
            icon={<FavoriteBorder />}
            onClick={() => navigate('/favorit')}
            data-testid="button-favorit"
          />
        )}
        {user?.role === 'user' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_cart_header' })}
            icon={<ShoppingCartOutlined />}
            onClick={() => navigate('/cart')}
            data-testid="button-cart"
          />
        )}
        {user?.role === 'merchant' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_services_header' })}
            icon={<DryCleaningOutlined />}
            onClick={() => navigate('/service')}
            data-testid="button-service"
          />
        )}
        {user?.role === 'merchant' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_merchant_header' })}
            icon={<LocalLaundryServiceOutlined />}
            onClick={() => navigate('/laundry')}
            data-testid="button-laundry"
          />
        )}

        {user?.role !== 'admin' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_order_list' })}
            icon={<LibraryBooksOutlined />}
            onClick={() => {
              if (user?.role === 'user') navigate('/user/order');
              else navigate('/laundry/orders');
            }}
            data-testid="button-order"
          />
        )}
        {user?.role !== 'admin' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_chat' })}
            icon={<MessageOutlined />}
            onClick={() => navigate('/chat')}
            data-testid="button-chat"
          />
        )}
        {user?.role === 'admin' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_user_page_header' })}
            icon={<PersonOutline />}
            onClick={() => navigate('/admin/user')}
            data-testid="button-admin"
          />
        )}
        {user?.role === 'admin' && (
          <BottomNavigationAction
            label={formatMessage({ id: 'app_deletedMerchants_header' })}
            icon={<LayersClear />}
            onClick={() => navigate('/admin/deletedMerchant')}
            data-testid="button-deletedMerchant"
          />
        )}
      </BottomNavigation>
    </div>
  );
};

MobileNavbar.propTypes = {
  login: PropTypes.bool,
  user: PropTypes.object,
  intl: PropTypes.object,
};
export default injectIntl(MobileNavbar);
