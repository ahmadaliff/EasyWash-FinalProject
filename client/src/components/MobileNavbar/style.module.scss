@import '@styles/base';

.MobileNavbar {
  position: sticky;
  bottom: 0;
  z-index: 15;
  @include apply-at-sm-size {
    display: none;
  }
}
.MobileNavbar.hide {
  display: none;
}
