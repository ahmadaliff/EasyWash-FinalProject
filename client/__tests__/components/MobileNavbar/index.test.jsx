import { fireEvent, render } from '@utils/testHelper';
import MobileNavbar from '@components/MobileNavbar';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let navbarUser;
let navbarAdmin;
let navbarMerchant;

const mockProps = {
  login: true,
  intl: { formatMessage: jest.fn() },
};

describe('Mobile Navbar Component', () => {
  describe('Navbar with role User', () => {
    beforeEach(() => {
      navbarUser = render(
        <MobileNavbar
          {...mockProps}
          user={{
            role: 'user',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarUser;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should call navigate to favorit when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-favorit');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/favorit`);
    });
    test('Should call navigate to cart when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-cart');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/cart`);
    });
    test('Should call navigate to order when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-order');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/user/order`);
    });
    test('Should call navigate to chat when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-chat');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/chat`);
    });
  });
  describe('Navbar with role Admin', () => {
    beforeEach(() => {
      navbarAdmin = render(
        <MobileNavbar
          {...mockProps}
          user={{
            role: 'admin',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarAdmin;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should call navigate to admin user when button clicked', () => {
      const { getByTestId } = navbarAdmin;
      const button = getByTestId('button-admin');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/admin/user`);
    });
    test('Should call navigate to admin deleted merchants when button clicked', () => {
      const { getByTestId } = navbarAdmin;
      const button = getByTestId('button-deletedMerchant');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/admin/deletedMerchant`);
    });
    test('Should match with snapshot', () => {
      expect(navbarAdmin).toMatchSnapshot();
    });
  });
  describe('Navbar with role Merchant', () => {
    beforeEach(() => {
      navbarMerchant = render(
        <MobileNavbar
          {...mockProps}
          user={{
            role: 'merchant',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarMerchant;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should call navigate to order when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-order');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/laundry/orders`);
    });
    test('Should call navigate to service when button clicked', () => {
      const { getByTestId } = navbarMerchant;
      const button = getByTestId('button-service');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/service`);
    });
    test('Should call navigate to laundry when button clicked', () => {
      const { getByTestId } = navbarMerchant;
      const button = getByTestId('button-laundry');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/laundry`);
    });
  });
});
