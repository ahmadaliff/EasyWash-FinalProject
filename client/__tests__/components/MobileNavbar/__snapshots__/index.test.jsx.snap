// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Navbar Component Navbar with role Admin Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MobileNavbar false"
        data-testid="navbar"
      >
        <div
          class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
        >
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
            data-testid="button-admin"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
            >
              Manajemen User
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
            data-testid="button-deletedMerchant"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LayersClearIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
            >
              Laundry yang dihapus
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MobileNavbar false"
      data-testid="navbar"
    >
      <div
        class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
      >
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
          data-testid="button-admin"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Manajemen User
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
          data-testid="button-deletedMerchant"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LayersClearIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Laundry yang dihapus
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
